### PR TITLE
Multicast server and request support

### PR DESCRIPTION
**Pull request by mast**
Here multicast request and server support is added.
Related issue https://github.com/mcollina/node-coap/issues/6

### CODE REVIEW

**Comment by mast** [comment]
It's strange that Travis fails. All tests (both new and old) pass in my environment.

**Comment by neich** [comment]
Check your node version. Travis is failing with node v0.12, nut not with v0.10 ...

**Comment by mast** [comment]
@neich my node is 0.10.37, so I guess that's why I don't see any fails.

**Comment by mcollina** [comment]
You can use [nvm(https://github.com/creationix/nvm) to switch between multiple node versions.

**Comment by mast** [comment]
It seems that newer versions of node doesn't allow to reuse same port/address for 2 sockets. It forces me to remove multicast test cases related to 2 servers started on the same host.

File: lib/agent.js [hunk 1]
```diff
@@ -330,12 +342,21 @@ Agent.prototype.request = function request(url) {
...
     }
   })
 
+  req.multicastTimer = setTimeout(function() {
+    that._msgInFlight--
+    if (that._msgInFlight === 0) {
+      that._doClose()
+    }
+  }, multicastTimeout)
+
```
Comment (mcollina) [inline lib/agent.js]: This multicastTimer should be allocated only if it is a multicast request.

Comment (mast) [inline lib/agent.js]: It's very good point. Thank you for noticing!
I pushed the update.

**Comment by mcollina** [comment]
Perfect!

**Pull request merged**